src/ext/java/org/opentripplanner/ext/flex/trip/UnscheduledTrip.java

```diff
@@ -154,9 +158,15 @@ public Stream<FlexAccessTemplate> getFlexAccessTemplates(
       );
   }
 
+  @Nullable
+  @Override
+  public String logName() {
+    return "timePenalty=(%s)".formatted(timePenalty.toString());
```

Comment:
I do not think this is what the `logName()` method is intended for - it is to add name/description to entity logging. If the goal is to tests the penalty, the best way to do it is to test that the calculator retuned is of type  TimePenaltyCalculator - that should be suffisient, since I assume there is unit tests on the calculator - no need to test the same features twice. Theoretical this is a white-box testing, but it is better than nothing.

Reply:
I feel a bit stupid because I should have thought of that myself. :man_facepalming: